codeguru_profiler_agent/sdk_reporter/sdk_reporter.py

```diff
@@ -90,11 +97,64 @@ def report(self, profile):
             )
             logger.info("Reported profile successfully")
             return True
+        except ClientError as error:
+            if error.response['Error']['Code'] == 'ResourceNotFoundException':
+                if self.is_lambda_one_click_integration_active():
+                    global is_create_pg_called_during_submit_profile
+                    is_create_pg_called_during_submit_profile = True
+                    logger.info("Received ResourceNotFoundException. Attempting to create 1-click PG in Report Profile")
+                    self.create_pg_when_one_click_integration_is_active()
+            return False
         except Exception as e:
             self._log_request_failed(operation="post_agent_profile", exception=e)
             return False
 
+    @with_timer("createOneClickPG", measurement="wall-clock-time")
+    def create_pg_when_one_click_integration_is_active(self):
+        """
+        Create a PG for the Lambda function onboarded with 1-click integration
+        """
+        function_name = str(self.metadata.fleet_info.function_arn).split(':')[6]
+
+        profiling_group_name_for_1_click = "aws-lambda-{}".format(function_name)
+        profiling_group_name_to_be_created = os.getenv("AWS_CODEGURU_PROFILER_GROUP_NAME",
+                                                       profiling_group_name_for_1_click)
+        try:
+            self.codeguru_client_builder.codeguru_client.create_profiling_group(
+                profilingGroupName=profiling_group_name_to_be_created,
+                computePlatform='AWSLambda'
+            )
+            self.profiling_group_name = profiling_group_name_to_be_created
+            self.is_lambda_one_click_pg_created_during_execution = True
+            logger.info("Created Lambda Profiling Group with name {}", profiling_group_name_to_be_created)
+
+        except Exception as e:
+            self._log_request_failed(operation="create_profiling_group", exception=e)
+
+    def is_lambda_one_click_integration_active(self):
+        """
+        Check if the ComputeType is AWSLambda and if the environment
+        variables for Lambda Layer Profiling are set
+        """
+        fleet_info_as_map = self.metadata.fleet_info.serialize_to_map()
+        if 'computeType' in fleet_info_as_map:
+            if fleet_info_as_map['computeType'] == 'aws_lambda':
+                handler_name_ = os.environ.get(HANDLER_ENV_NAME_FOR_CODEGURU)
+                if not handler_name_:
+                    logger.error("Env Variables for CodeGuru Profiler Lambda Layer are not set. Cannot create PG.")
```

Comment:
Would this be printed if the customer is using Lambda, but without the layer integration?
* https://docs.aws.amazon.com/codeguru/latest/profiler-ug/python-lambda-command-line.html

If so, we should not log error then.

Reply:
no this is only when the customer is using Lambda layer. 1-click support is only for the layer I believe.